cmd/export_ledger_entry_changes.go

```diff
@@ -91,77 +97,120 @@ be exported.`,
 								continue
 							}
 							transformedOutputs["claimable_balances"] = append(transformedOutputs["claimable_balances"], balance)
+
+							if ok, actualLedger := utils.LedgerIsCheckpoint(entry.LastModifiedLedgerSeq); ok {
+								x := verifyOutputs[actualLedger]
+								x.Claimable_balances = append(x.Claimable_balances, balance)
+								verifyOutputs[actualLedger] = x
+							}
 						}
 					case xdr.LedgerEntryTypeOffer:
 						for _, change := range changes {
+							entry, _, _, _ := utils.ExtractEntryFromChange(change)
 							offer, err := transform.TransformOffer(change)
 							if err != nil {
-								entry, _, _, _ := utils.ExtractEntryFromChange(change)
 								cmdLogger.LogError(fmt.Errorf("error transforming offer entry last updated at %d: %s", entry.LastModifiedLedgerSeq, err))
 								continue
 							}
 							transformedOutputs["offers"] = append(transformedOutputs["offers"], offer)
+
+							if ok, actualLedger := utils.LedgerIsCheckpoint(entry.LastModifiedLedgerSeq); ok {
+								x := verifyOutputs[actualLedger]
+								x.Offers = append(x.Offers, offer)
+								verifyOutputs[actualLedger] = x
+							}
 						}
 					case xdr.LedgerEntryTypeTrustline:
 						for _, change := range changes {
+							entry, _, _, _ := utils.ExtractEntryFromChange(change)
 							trust, err := transform.TransformTrustline(change)
 							if err != nil {
-								entry, _, _, _ := utils.ExtractEntryFromChange(change)
 								cmdLogger.LogError(fmt.Errorf("error transforming trustline entry last updated at %d: %s", entry.LastModifiedLedgerSeq, err))
 								continue
 							}
 							transformedOutputs["trustlines"] = append(transformedOutputs["trustlines"], trust)
+
+							if ok, actualLedger := utils.LedgerIsCheckpoint(entry.LastModifiedLedgerSeq); ok {
+								x := verifyOutputs[actualLedger]
+								x.Trustlines = append(x.Trustlines, trust)
+								verifyOutputs[actualLedger] = x
+							}
 						}
 					case xdr.LedgerEntryTypeLiquidityPool:
 						for _, change := range changes {
+							entry, _, _, _ := utils.ExtractEntryFromChange(change)
 							pool, err := transform.TransformPool(change)
 							if err != nil {
-								entry, _, _, _ := utils.ExtractEntryFromChange(change)
 								cmdLogger.LogError(fmt.Errorf("error transforming liquidity pool entry last updated at %d: %s", entry.LastModifiedLedgerSeq, err))
 								continue
 							}
 							transformedOutputs["liquidity_pools"] = append(transformedOutputs["liquidity_pools"], pool)
+
+							if ok, actualLedger := utils.LedgerIsCheckpoint(entry.LastModifiedLedgerSeq); ok {
+								x := verifyOutputs[actualLedger]
+								x.Liquidity_pools = append(x.Liquidity_pools, pool)
+								verifyOutputs[actualLedger] = x
+							}
 						}
 					}
 				}
 
+				for checkpointLedgers := range verifyOutputs {
+					_, err := verify.VerifyState(ctx, verifyOutputs[checkpointLedgers], archive, checkpointLedgers, verifyBatchSize)
```

Comment:
Do you know where the `verifyBatchSize` comes from? Do we know that 50000 is big enough?

Reply:
`verifyBatchSize` value came from [here](https://github.com/stellar/go/blob/master/services/horizon/internal/ingest/verify.go#L23). About the second question, I don't know the answer. Maybe @bartekn could you help us with that?

Reply:
This value can be changed. It's basically as big as possible but small enough to not cause OOM crash.